cores/esp32/esp32-hal-uart.c

```diff
@@ -96,14 +110,14 @@ static void IRAM_ATTR _uart_isr(void *arg)
 void uartEnableInterrupt(uart_t* uart)
 {
     UART_MUTEX_LOCK();
-    uart->dev->conf1.rxfifo_full_thrhd = 112;
-    uart->dev->conf1.rx_tout_thrhd = 2;
+    uart->dev->conf1.rxfifo_full_thrhd = 112; // needs speed compensation?
+    uart->dev->conf1.rx_tout_thrhd = 32; // no need to force rxFiFo empty, because uartRead() will empty if necessary
     uart->dev->conf1.rx_tout_en = 1;
     uart->dev->int_ena.rxfifo_full = 1;
     uart->dev->int_ena.frm_err = 1;
     uart->dev->int_ena.rxfifo_tout = 1;
     uart->dev->int_clr.val = 0xffffffff;
-
+// can this share an interrupt?
```

Comment:
why?

Reply:
I had to share interrupts in I2C because of interrupt exhaustion problems identified at Release 1.0.0.  If it is possible to share interrupts then this change would free two.  

Maybe I'm miss-informed, is the default shared or individual?  When I specified `ESP_INTR_FLAG_EDGE` I created a conflict with a pin interrupt or timer. I had to explicitly ask for a shared interrupt with:
```
uint32_t flags = ESP_INTR_FLAG_IRAM |  //< ISR can be called if cache is disabled
          ESP_INTR_FLAG_LOWMED |   //< Low and medium prio interrupts. These can be handled in C.
          ESP_INTR_FLAG_SHARED; //< Reduce resource requirements, Share interrupts
```

I may be barking at a problem that does not exist.
 
I just see you only specifying: `ESP_INTR_FLAG_IRAM` without any shared option.
```
esp_intr_alloc(UART_INTR_SOURCE(uart->num), (int)ESP_INTR_FLAG_IRAM, _uart_isr, NULL, &uart->intr_handle);
```

Reply:
I vote for trying and also very important... let's change the arg in `esp_intr_alloc` to not be null but instead to point to the uart struct so loop through all interfaces will not be necessary when ISR fires.